extensions/interactions/ItemSelectionInput/directives/ItemSelectionInputValidationService.js

```diff
@@ -177,8 +180,31 @@ oppia.factory('ItemSelectionInputValidationService', [
                       'rule must include at least 2 options.')
                   });
                 }
+              } else if (rule.type === 'Equals') {
+                if (minAllowedCount > ruleInputs.length) {
+                  warningsList.push({
+                    type: WARNING_TYPES.ERROR,
+                    message: (
+                      'In answer group ' + (answerIndex + 1) + ', ' +
+                      'rule ' + (ruleIndex + 1) + ', the "Equals" rule ' +
+                      'should not have more correct options than minimum ' +
+                      'allowed selection count.')
+                  });
+                }
               }
             });
+            if (ruleInputs.length === 0) {
```

Comment:
Ditto. Check for both minAllowedCount and maxAllowedCount.

Reply:
I did not understand this part, As for the maxAllowed and minAllowed, Yeah I will change it now.

Reply:
As it is `ContainsAtLeastOneOf ` rule, so we can refrain from checking the bounds on ruleInputs.length. Please ignore my previous comment!